### PR TITLE
feat(#854): Add `getServerSession` and `getToken` to the local provider

### LINKED ISSUE #854: Add server-side access to tokens and session for the `local` and `refresh` providers
### Describe the feature

A set of simple server-side functions that mirror `getServerSession` and `getToken` from the `authjs` provider, but for the `local` and `refresh` providers. 

### How would you implement this?

Add the two new functions, extract the token via the cookie from the h3 `event`. To retrieve the session data, use the token and send a backend request to the configured authentication API and return the data.

### Additional information

- [X] Would you be willing to help implement this feature?

### Provider

- [ ] AuthJS
- [X] Local
- [X] Refresh
- [ ] New Provider

### PR DESCRIPTION
**Pull request by zoey-kaiser**
### 🔗 Linked issue

Closes #854

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [X] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Adds `getServerSession` and `getToken` to the server-side ultis for the `local` provider. 

### 📝 Checklist

- [X] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/utils/generateTypes.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { createResolver } from '@nuxt/kit'
+import type { AuthProviders } from '../types'
+
+export function generateModuleTypes(provider: AuthProviders) {
+  const { resolve } = createResolver(import.meta.url)
+
+  const providerSpecificTypes: string[] = []
+
+  if (provider.type === 'authjs') {
+    providerSpecificTypes.push(`  const { getServerSession, getToken, NuxtAuthHandler }: typeof import('${resolve('./runtime/server/services/authjs')}')`)
```
Comment (phoenix-ru) [inline src/runtime/utils/generateTypes.ts:11]: Question - does this resolve relative to current path or from root? From the docs, it looks like the resolution will happen from the current path which is not correct:
https://nuxt.com/docs/api/advanced/import-meta#builder-properties

File: src/runtime/server/services/local/getToken.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+export function getToken(event: H3Event) {
+  const config = useTypedBackendConfig(useRuntimeConfig(), 'local')
+  const rawToken = getCookie(event, config.token.cookieName)
+  const token = formatToken(rawToken, config)
+
+  if (!token) {
+    return null
+  }
+  return token
```
Comment (phoenix-ru) [inline src/runtime/server/services/local/getToken.ts:14]: `formatToken` already returns `string | null`
```suggestion
  return token
```

File: src/runtime/server/services/local/getToken.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { type H3Event, getCookie } from 'h3'
+import { useTypedBackendConfig } from '../../../helpers'
+import { formatToken } from '../../../utils/local'
+import { useRuntimeConfig } from '#imports'
+
+export function getToken(event: H3Event) {
```
Comment (phoenix-ru) [inline src/runtime/server/services/local/getToken.ts:6]:
```suggestion
export function getToken(event: H3Event): string | null {
```

File: src/runtime/server/services/local/getServerSession.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import getURL from 'requrl'
+import { joinURL } from 'ufo'
+import { jsonPointerGet, useTypedBackendConfig } from '../../../helpers'
+import { getToken } from './getToken'
+
+// @ts-expect-error - #auth not defined
+import type { SessionData } from '#auth'
+import { useRuntimeConfig } from '#imports'
+
+function joinPathToApiURL(event: H3Event, path: string) {
```
Comment (phoenix-ru) [inline src/runtime/server/services/local/getServerSession.ts:11]: nit: would prefer if this was down the file as it is a helper (consumers are more interested in reading `getServerSession`

File: src/runtime/server/services/local/getServerSession.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  const { origin, pathname, fullBaseUrl } = useRuntimeConfig().public.auth.computed
+
+  let baseURL
+  if (origin) {
+    // Case 1: An origin was supplied by the developer in the runtime-config. Use it by returning the already assembled full base url that contains it
+    baseURL = fullBaseUrl
+  }
+  else {
+    // Case 2: An origin was not supplied, we determine it from the request
```
Comment (phoenix-ru) [inline src/runtime/server/services/local/getServerSession.ts:20]: This probably needs to happen only when `trustHost` is set (look at `authjs` implementation)

**Comment by zoey-kaiser** [comment]
Closing. Will revisit when #913 is merged.

**Pull request closed without merging**